packages/dashboard-core-plugins/src/panels/ConsolePanel.tsx

```diff
@@ -298,9 +298,9 @@ export class ConsolePanel extends PureComponent<
     assertNotNull(sessionWrapper);
 
     const { config, session } = sessionWrapper;
-    const { title } = widget;
-    assertNotNull(title);
-    const panelId = this.getItemId(title);
+    const { name } = widget;
+    assertNotNull(name);
+    const panelId = this.getItemId(name);
```

Comment:
We want to use `title` if it's available, then fallback to `name` if `title` is not available.

Other objects we open are actually a `VariableDefinition` type, and the `title` is the friendly name for the object and what we want to display to the user.

Reply:
Right, but then would it not make sense to pass in` title: title` instead of `name: title `? Given the declaration below, I was under the impression that the intention **was to pass in the `name`**? Since, even if I add a fallback to `name` if `title` is not available, it would not matter since `title` will never be available, given that the `title` is assigned to `name` here

https://github.com/deephaven/web-client-ui/blob/139388d77f94f561938bb4ce61da4780387ef4a9/packages/console/src/Console.tsx#L787

Reply:
You'd get a type error then, because a `VariableDescriptor` is required in `openWidget`.
The `title` vs. `name` thing is kind of annoying and due to the types provided from the JS API which we're not totally abstracted from here. We should try and use `title` if available, otherwise fallback to `name`.